source/common/http/conn_manager_impl.h

```diff
@@ -503,21 +520,27 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
       State() : remote_complete_(false), local_complete_(false), saw_connection_close_(false) {}
 
       uint32_t filter_call_state_{0};
+      bool encoder_filters_streaming_{true};
```

Comment:
Sorry I think we are talking past each other on this state stuff. The only reason the below booleans are a bitfield is to save space. It has nothing to do with mutual exclusion or flag like variables. For this reason any reason to not just define the three new variables as part of the bitfield?

Reply:
bool& is not exchangeable for bitfields in arguments.

destroyed_ is handed to sendLocalReply, and apparently the const bool& is good enough to keep the compiler happy.  Unfortunately it doesn't actually *work* - the const convinces the compiler to convert the bitfield to a boolean, but if you change the value during header serialization it just fails to pick up the edit.  Highly related: I'm so glad I wrote (and deleted) a unit test for this one.

  EXPECT_CALL(callbacks, encodeHeaders_(_, false)).WillOnce(InvokeWithoutArgs([&]() -> void {
    bitfield.destroyed_ = true;  // faails to
  }));
  EXPECT_CALL(callbacks, encodeData(_, true)).Times(0);  // unexpectedly called, test fails.

For commonHandleAfterDataCallback the bitfield/bool conversion fails outright.  This one we could arguably refactor to take the State struct instead of the specific boolean it edits, I just prefer the clarity of passing the argument to eb edited..  Let me know if you prefer the compactness and I'll get right on it :-)

Reply:
Yikes OK. TIL something new. It's not a big deal either way. I guess I would say pick between one of these:
1) Just get rid of bitfield entirely in there for consistency.
2) Add a comment on why some are bitfield and some are not.
3) Refactor to pass state around as you mentioned.